### PR TITLE
add query details to query view page

### PR DESCRIPTION
**Pull request by eriktaubeneck**
Provides more detail to the query view page:

image

## Summary by CodeRabbit

- **New Features**
  - Enhanced the query page to dynamically display query information.
  - Introduced state management for fetching and displaying query details.
  - Added functionality to handle checkbox events based on query data.

### CODE REVIEW

File: server/app/query/view/[id]/page.tsx [hunk 1]
```diff
@@ -86,18 +89,98 @@ export default function QueryPage({ params }: { params: { id: string } }) {
...
+        <div>
+          <button
+            onClick={() => kill(IPARemoteServers)}
+            type="button"
+            className="ml-3 rounded-md bg-red-600 px-3 py-2 text-sm font-semibold text-white shadow-sm hover:bg-red-700 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-red-600"
+          >
+            Kill Query
+          </button>
+        </div>
+      </div>
```
[the remark this answers is not in the document]
Comment (eriktaubeneck) [reply server/app/query/view/[id]/page.tsx:107]: lol it's already there.

File: server/app/query/view/[id]/page.tsx [hunk 2]
```diff
@@ -86,18 +89,98 @@
...
+                      key={key}
+                    >
+                      <div className="flex w-0 flex-1 items-center">
+                        <div className="ml-4 flex min-w-0 flex-1 gap-2">
+                          <span className="truncate font-medium"> {key}</span>
+                        </div>
+                      </div>
+                      <div className="ml-4 flex-shrink-0">
+                        <a
+                          href="#"
```
[the remark this answers is not in the document]
Comment (eriktaubeneck) [reply server/app/query/view/[id]/page.tsx]: good catch, this shouldn't be a link.

File: server/app/query/view/[id]/page.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (akoshelev) [reply server/app/query/view/[id]/page.tsx:107]: don't make it smarter @eriktaubeneck , you risk the future for all of us

File: server/app/query/view/[id]/page.tsx [hunk 3]
```diff
@@ -86,18 +89,90 @@ export default function QueryPage({ params }: { params: { id: string } }) {
...
-        </h2>
+      <div className="flex justify-between items-center">
+        <div>
+          <h2 className="text-2xl font-bold leading-7 text-gray-900 dark:text-gray-100 sm:truncate sm:text-3xl sm:tracking-tight">
+            Query Details
+          </h2>
+        </div>
+        <div>
+          <button
+            onClick={() => kill(IPARemoteServers)}
```
Comment (akoshelev) [inline server/app/query/view/[id]/page.tsx:100]: I am not up to date with the best practices for developing UIs, is it the recommended way to execute actions from the HTML markup?

File: server/app/query/view/[id]/page.tsx [hunk 4]
```diff
@@ -86,18 +89,90 @@ export default function QueryPage({ params }: { params: { id: string } }) {
...
+            </dd>
+          </div>
+
+          <div className="px-4 py-2 sm:grid sm:grid-cols-3 sm:gap-4 sm:px-0">
+            <dt className="text-sm font-medium leading-6 text-gray-900">
+              Params:
+            </dt>
+            <dd className="mt-2 text-sm text-gray-900 sm:col-span-2 sm:mt-0">
+              <ul className="divide-y divide-gray-200 rounded-md border border-gray-200">
+                {Object.entries(
```
Comment (akoshelev) [inline server/app/query/view/[id]/page.tsx:153]: again this seems to be quite involved and maybe require some logging in the future, is there a way to do it in JS and map to UI here

**Review by akoshelev — APPROVED** [review]

File: server/app/query/view/[id]/page.tsx (hunk 4 above)
Comment (eriktaubeneck) [reply server/app/query/view/[id]/page.tsx:153]: good suggestion to move some of this out of the template. that said, the fun thing about react nowadays is you could add logging now a couple lines down (just before the return statement). I had one there as I was developing actually!

File: server/app/query/view/[id]/page.tsx (hunk 3 above)
Comment (eriktaubeneck) [reply server/app/query/view/[id]/page.tsx:100]: yep! react is basically php but in JavaScript (not surprising since it came out of FB...)

**Pull request merged**